public/static/docs/user-guide/setup-google-drive-remote.md

```diff
@@ -0,0 +1,96 @@
+   account.
+2. Create a **New Project** or select an available one.
+3. From the left sidebar menu select the **APIs & Services** -> **Dashboard**
+   and click at the **ENABLE APIS AND SERVICES** button to type and search for
+   the **Google Drive API** to enable it from the search results.
+
+   ![](/static/img/gdrive-enable-apis-and-services.png)
+
+4. From the left sidebar menu select the **APIs & Services** -> **OAuth consent
+   screen**, select a desired **User Type** and click the **Create** button. On
+   the next screen enter an **Application name** and click the **Save** button.
+   You will be able to back and edit others settings later.
+
+5. From the left sidebar menu select the **APIs & Services** -> **Credentials**
+   and click the **Create credentials** button to select the **OAuth client ID**
+   -> **Other** -> click **Create** to proceed with a default suggested **Name**
+   for your new **OAuth client ID**.
+
+   ![](/static/img/gdrive-create-credentials.png)
+
+6. The new generated **client ID** and **client secret** should be showed to
+   you. Use them for further DVC's configuration.
+
+```dvc
+$ dvc remote add -d myremote gdrive://path # See URL format section below
+$ dvc remote modify myremote gdrive_client_id my_gdrive_client_id
+$ dvc remote modify myremote gdrive_client_secret gdrive_client_secret
+```
+
+⚠️ It is safe to share a **client ID** and a **client secret** at least with
+your teammates or inside an organization. These credentialds are used to
+[retrieve an authorization token](https://developers.google.com/identity/protocols/OAuth2)
+and
+[authenticate your application](https://developers.google.com/drive/api/v2/about-auth)
+to Google. Credentials don't give an access to your Google Drive storage. But
+Google limits global number of queries per each **client ID** and uncontrolled
+usage of your **client ID** by outer world can lead to
+[reaching the allowed Google API usage limits](https://developers.google.com/drive/api/v2/handle-errors?hl=ro#quota).
+
+## Google Drive remote URL format
+
+A remote URL can contain **root** which is an alias of your topmost Google Drive
+directory. To point the remote to the path under a Google Drive's **root**
+directory:
+
+```dvc
+$ dvc remote add -d myremote gdrive://root/path/to/dir
+```
+
+As alternative, you can obtain unique **Google Drive id** of your desired remote
+directory and use it as **base** remote path in URL:
+
+```dvc
+$ dvc remote add -d myremote gdrive://0AIac4JZqHhKmUk9PDA/path/to/dir
+```
+
+To obtain the **Google Drive ID** of a remote directory navigate inside that
+directory in your web browser with opened Google Drive and find the **Google
+Drive ID** in address bar. For example, for the URL
+
+```dvc
+https://drive.google.com/drive/folders/0AIac4JZqHhKmUk9PDA
+```
+
+the **ID** will be equal to `0AIac4JZqHhKmUk9PDA`.
+
+**Shared Team Drive** can be referenced by **Google Drive ID** in the similar
+way.
```

Comment:
OK guys (@MaxRis) this is probably the only part that still needs some conceptial review. I'm rewriting it a little bit but TBH I still find it somewhat confusing:
- What is a `root`? It says "your topmost Google Drive directory" but what does this look like? An example is probably needed.
- "As alternative, you can obtain unique **Google Drive id**" As an alternative to `root`, or to `path/to/dir`? The example below is `gdrive://0AIac4JZqHhKmUk9PDA/path/to/dir` so does this Gdrive _Folder ID_ belong to the aforementioned `root`? (Otherwise I'll remove the `path/to/dir` part from this 2nd example.)

Reply:
`root` - is a predefined name. If you don't use an ID like `AssnderQW24w23` you must always use the prefix `root`

Reply:
Oh so it's literally "root"? OK got it, I'll make it more explicit. I need to actually try all these instructions now that I've reviewed the language... 🙂

Reply:
URL is constructed from "base" path and optional path/to/dir.
gdrive://base/path/to/dir

Where the "base" can be:

1. root
2. ID of folder
3. appDataFolder

@jorgeorpinel I'm sorry, I've completely forgot that we also support `appDataFolder`. It is alias similar to "root", but it points to the hidden for users directory used implicitly by DVC application. Here it is more info about it https://developers.google.com/drive/api/v2/appdata
Please let me know if you will able to add mentioning of `appDataFolder` as well, or if my help is needed here, thanks!

Reply:
OK thanks, I'll add this info.

Reply:
OK I rewrote this section now, clarifying and adding the missing info. Haven't actually tried all the options though... Will do that ASAP.